src/main/java/com/nekomaster1000/infernalexp/mixin/common/MixinAbstractFireBlock.java

```diff
@@ -1,29 +1,58 @@
 package com.nekomaster1000.infernalexp.mixin.common;
 
-import net.minecraft.block.AbstractBlock;
-import net.minecraft.block.AbstractFireBlock;
-import net.minecraft.block.Block;
-import net.minecraft.block.BlockState;
-import net.minecraft.util.math.BlockPos;
-import net.minecraft.world.IBlockReader;
-
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
+import com.nekomaster1000.infernalexp.access.SoulFireAccess;
+import com.nekomaster1000.infernalexp.access.SoulFireAccess.KnownFireTypes;
 import com.nekomaster1000.infernalexp.blocks.GlowFireBlock;
 import com.nekomaster1000.infernalexp.init.IEBlocks;
 
+import net.minecraft.block.AbstractBlock;
+import net.minecraft.block.AbstractFireBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.ModList;
+
 @Mixin(AbstractFireBlock.class)
-public class MixinAbstractFireBlock extends Block {
-        
-        private MixinAbstractFireBlock(AbstractBlock.Properties properties) {
-            super(properties);
-        }
-    
-        @Inject(method = "getFireForPlacement", at = @At("HEAD"), cancellable = true)
-        private static void getFireForPlacement(IBlockReader reader, BlockPos pos, CallbackInfoReturnable<BlockState> info) {
-            if (GlowFireBlock.isGlowFireBase(reader.getBlockState(pos.down()).getBlock())) info.setReturnValue(IEBlocks.GLOW_FIRE.get().getDefaultState());
-        }
-}
\ No newline at end of file
+public abstract class MixinAbstractFireBlock extends Block {
+
+	private MixinAbstractFireBlock(AbstractBlock.Properties properties) {
+		super(properties);
+	}
+
+	@Inject(method = "getFireForPlacement", at = @At("HEAD"), cancellable = true)
+	private static void IE_getFireForPlacement(IBlockReader reader, BlockPos pos, CallbackInfoReturnable<BlockState> info) {
+		if (GlowFireBlock.isGlowFireBase(reader.getBlockState(pos.down()).getBlock())) {
+			info.setReturnValue(IEBlocks.GLOW_FIRE.get().getDefaultState());
+		}
+	}
+
+	@Inject(method = "onEntityCollision", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/Entity;forceFireTicks(I)V"))
+	private void IE_setCustomFires(BlockState state, World worldIn, BlockPos pos, Entity entityIn, CallbackInfo info) {
+		SoulFireAccess access = ((SoulFireAccess) entityIn);
+		if (state.isIn(Blocks.SOUL_FIRE)) {
+			access.setFireType(KnownFireTypes.SOUL_FIRE);
+		} else if (state.isIn(IEBlocks.GLOW_FIRE.get())) {
+			access.setFireType(KnownFireTypes.FIRE_GLOW);
```

Comment:
```suggestion
			access.setFireType(KnownFireTypes.GLOW_FIRE);
```

why does glow fire get a different naming scheme

Reply:
i dont know